test/livebook_web/live/session_live/secrets_component_test.exs

```diff
@@ -72,8 +59,127 @@ defmodule LivebookWeb.SessionLive.SecretsComponentTest do
       render_change(form, attrs)
       render_submit(form, attrs)
 
+      assert_receive {:secret_created, ^secret}
       assert render(view) =~ "A new secret has been created on your Livebook Enterprise"
       assert has_element?(view, "#hub-#{enterprise.id}-secret-#{attrs.data.name}-title")
+
+      assert has_element?(
+               view,
+               "#hub-#{enterprise.id}-secret-#{secret.name}-title span",
+               enterprise.hub_emoji
+             )
+    end
+
+    test "toggle a secret from Enterprise hub",
+         %{conn: conn, session: session, enterprise: enterprise, node: node} do
+      secret =
+        build(:secret,
+          name: "POSTGRES_PASSWORD",
+          value: "postgres",
+          origin: {:hub, enterprise.id}
+        )
+
+      {:ok, view, _html} = live(conn, Routes.session_path(conn, :page, session.id))
+
+      :erpc.call(node, Enterprise.Integration, :create_secret, [secret.name, secret.value])
+      assert_receive {:secret_created, ^secret}
+
+      Session.set_secret(session.pid, secret)
+      assert_session_secret(view, session.pid, secret)
+    end
+
+    test "adding an unavailable secret using 'Add secret' button",
```

Comment:
what does it mean for secret to be unavailable?

Reply:
When you run `System.fetch_env!("LB_" <> _)` (with `LB_` prefix), it raises an exception. We get this exception with `Livebook.Runtime.Evaluator.DefaultFormatter` ([here](https://github.com/livebook-dev/livebook/blob/dd68f7ff1c2632c4d37a66a104717556b7acb4a5/lib/livebook/runtime/evaluator/default_formatter.ex#L139-L140)) and render the error + "Add button" with `LivebookWeb.Output` ([here](https://github.com/livebook-dev/livebook/blob/8f926619186d35d0b73eda2eb452c2c86ec1853e/lib/livebook_web/live/output.ex#L249-L280)).

Then, after clicking the button, it opens the Secrets modal with `prefilled_secret_name`, which would show the `Grant access` banner + button and the Available secrets list.

Reply:
Ok that’s pretty amazing. Maybe s/unavailable/missing since we use the latter name in code?